website/docs/cloud-docs/api-docs/state-versions.mdx

```diff
@@ -164,6 +172,26 @@ curl \
 }
 ```
 
+## Upload State and JSON State
+
+-> **Note:** While you can upload state version content in the same request when creating a state version, we strongly recommended that you upload content directly to the provided URL.
+
+`PUT https://archivist.terraform.io/v1/object/<UNIQUE OBJECT ID>`
```

Comment:
do they need to get the object id? where do they reference the URL?

Reply:
The entire URL is provided `hosted-state-upload-url` or `hosted-json-state-upload-url` field, so they don't need to fetch any object ID.